File: solutions/java/high-scores/1/src/main/java/HighScores.java
import java.util.List;
import java.util.ArrayList;
import java.util.Collections;

class HighScores {

    private List<Integer> highScores;

    public HighScores(List<Integer> highScores) {
        this.highScores = new ArrayList<>(highScores);
    }

    List<Integer> scores() {
        return new ArrayList<>(highScores);
    }

    Integer latest() {
        return highScores.get(highScores.size() - 1);
    }

    Integer personalBest() {
        return Collections.max(highScores);
    }

    List<Integer> personalTopThree() {
        List<Integer> sortedScores = new ArrayList<>(highScores);
        Collections.sort(sortedScores, Collections.reverseOrder());
        return sortedScores.subList(0, Math.min(3, sortedScores.size()));
    }

}

File: solutions/cpp/high-scores/1/high_scores.cpp
#include "high_scores.h"
#include <algorithm>
namespace arcade {
std::vector<int> HighScores::list_scores() {
    return scores;
}
int HighScores::latest_score() {
    return scores.back();
}
int HighScores::personal_best() {
    std::vector<int>::iterator max =std::max_element(scores.begin(), scores.end());
    return *max;
}
std::vector<int> HighScores::top_three() {
    std::vector<int> copy{scores};
    std::sort(copy.begin(), copy.end(), std::greater<int>());
    if (copy.size()<3){return copy;}
    return std::vector<int>(copy.begin(), copy.begin() + 3);
}
}  // namespace arcade
